src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -474,7 +482,11 @@ def __init__(
             self._schema_query = sql
             self._query_param = params
 
-    def __deepcopy__(self, memodict={}) -> "SelectSQL":  # noqa: B006
+    def __deepcopy__(self, memodict=None) -> "SelectSQL":  # noqa: B006
```

Comment:
@sfc-gh-aalam a clarify question so the memodict is used to avid repeated deepcopy of the same node, right? for example if a node whose id exists in he meodict, it will not repeated copy the node, is the correct?

Reply:
for other nodes that are not selectable and snowflakeplane, do they atuomatically added it to memodict? and check the memodict?

Reply:
We avoid deepcopy, YES. But also, since we have already seen the old object, we will return the same old deep copied object.

Reply:
For other plans, we can call `deepcopy` with and without `memodict`. If it is called without a memodict, the internal implementation with create a new memodict. Otherwise, we use the memodict passed by the user.